blog/2025-10-22-help-us-test-opentofu-1-11-0-beta1.md

```diff
@@ -0,0 +1,245 @@
+| macOS (Intel)         | [tofu_1.11.0-beta1_darwin_amd64.tar.gz](https://github.com/opentofu/opentofu/releases/download/v1.11.0-beta1/tofu_1.11.0-beta1_darwin_amd64.tar.gz) |
+| Linux (AMD64)         | [tofu_1.11.0-beta1_linux_amd64.tar.gz](https://github.com/opentofu/opentofu/releases/download/v1.11.0-beta1/tofu_1.11.0-beta1_linux_amd64.tar.gz)   |
+| Linux (ARM64)         | [tofu_1.11.0-beta1_linux_arm64.tar.gz](https://github.com/opentofu/opentofu/releases/download/v1.11.0-beta1/tofu_1.11.0-beta1_linux_arm64.tar.gz)   |
+
+After downloading, unpack the archive to find your `tofu` binary. For verified downloads, you can also use the [standalone installer](https://opentofu.org/docs/intro/install/standalone/) with signature verification.
+
+## What's New in 1.11.0-beta1
+
+### Ephemeral Resources - Beta Release
+
+Ephemeral resources allow you to work with sensitive data, temporary credentials, and transient infrastructure without persisting them to your state.
+
+A long-standing challenge with infrastructure-as-code has been the storage of sensitive data within state. This includes passwords, API keys, certificates, and other secrets that could lead to security incidents if leaked. Ephemeral resources solve this problem by ensuring these values only exist during the execution of a single OpenTofu command.
+
+#### Key Benefits
+
+- **Enhanced Security**: Sensitive values never touch your state files
+- **Transient Resources**: Create temporary network tunnels, SSH proxies, or time-limited credentials
+- **Cleaner State Management**: Reduce the amount of sensitive data requiring encryption and protection
+
+#### How It Works
+
+Ephemeral resources are opened at the beginning of an OpenTofu operation, used during execution, and closed when no longer needed. No ephemeral values are ever stored in your state.
+
+Here's a simple example that also uses write-only attributes; These allow you to pass ephemeral data into regular resources without storing that information in your state:
+
+```hcl
+# Use AWS SecretsManager to generate a random password
+ephemeral "aws_secretsmanager_random_password" "password" {
+
+}
+
+resource "kubernetes_secret_v1" "credentials" {
+  metadata = {
+    name = "admin"
+    namespace = "my-app"
+  }
+  data_wo = {
+    username = "admin"
+    password = ephemeral.aws_secretsmanager_random_password.password.random_password
+  }
+
+  data_wo_revision = 1
+  type = "kubernetes.io/basic-auth"
+}
+```
+
+In this case, we're using the `aws_secretsmanager_random_password` ephemeral resource to generate a random password and store it in a kubernetes secret.
+
+The value of this password will never be stored in the plan or the state by OpenTofu. Because OpenTofu does not store the value in the state the resource will only update it's password value when the revision number is incremented.
+
+You can find more in our documentation for [Ephemerality](/docs/main/language/ephemerality/ephemeral-resources/)
```

Comment:
I would suggest to have a mention about write-only arguments too. Saying this because it might create confusion about the availability of it, as it could have been the case in other projects.